drivers/w1/w1_ds2477_85_common.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2022 Thomas Stranger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Common functions for Maxim DS2477,DS2485 1-Wire Masters
+ */
+
+#include "w1_ds2477_85_common.h"
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/w1.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+
+LOG_MODULE_REGISTER(w1_ds2477_85, CONFIG_W1_LOG_LEVEL);
+
+int ds2477_85_write_port_config(const struct device *dev, uint8_t reg, uint16_t value)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	uint8_t buf[5] = {CMD_WR_W1_PORT_CFG, CMD_WR_W1_PORT_CFG_LEN, reg};
+	int ret;
+
+	__ASSERT_NO_MSG(reg <= PORT_REG_COUNT);
+
+	sys_put_le16(value, &buf[3]);
+	ret = i2c_write_dt(&cfg->i2c_spec, buf, (CMD_WR_W1_PORT_CFG_LEN + CMD_OVERHEAD_LEN));
+	if (ret < 0) {
+		return -EIO;
+	}
+
+	k_usleep(cfg->t_op_us);
+
+	ret = i2c_read_dt(&cfg->i2c_spec, buf, 2);
+	if ((ret < 0) || (buf[0] != 1) || (buf[1] != DS2477_88_RES_SUCCESS)) {
+		return -EIO;
```

Comment:
Just wondering if it is possible to return `ret` here instead of `-EIO`?

Reply:
I'd have to introduce a second `if()` for the buffer checks, and wouldn't change much because I2c_read is expected to return either `0` or `-EIO`.

Reply:
If now changed this and all similar occurrences as well.

Reply:
I think this is the wrong order. Currently it first checks the content and afterwards if the transaction was successful. Not dramatic in this case but might shadow the actual error from the I2C transaction.

Reply:
I have now restored the order. Thanks.